Tests

iros opened the pull request:
Test infrastructure + a few test files, one for unit tests (using formatters) and one testing the actual react component (taco table).
Uses Karma + Mocha + Chai, and the Chrome karma tester.

iros committed:
starting tests

iros committed:
added a few formatte tests, eslint rules and eslint fixes

iros committed:
Merge branch 'master' into tests

Conflicts:
	package.json

iros committed:
Fixing hot reloading issue & adding tests for taco table sorting

iros committed:
Added test instructions to readme

pbeshai commented on tests/Formatters.js:
```diff
@@ -0,0 +1,41 @@
+      assert.equal(Formatters.safeFormat(undefined, identity), undefined);
+    });
+
+    it('should return value when passed a value that is safe', function () {
```
should return at **formatted** value

iros replied:
Updating all test names to reflect where a formatted value should be returned.

pbeshai commented on tests/TacoTable.js:
```diff
@@ -0,0 +1,133 @@
+    return findRenderedComponentWithType(table, TacoTable);
+  }
+
+  const columnDefinition = [
```
columnDefinition**s** or simply **columns**

iros replied:
+1 columns.

pbeshai commented on tests/TacoTable.js:
```diff
@@ -0,0 +1,133 @@
+      assert.equal(foundTable.props.columns.length, 2);
+    });
+    it('should have 5 rows', function () {
+      assert.equal(foundTable.props.data.length, 5);
```
These tests seem to only check that the props was set, but really we'd like to see if there are 5 TRs in the <tbody> with 2 <TD>s in each

iros replied:
Adding actual dom tests.

iros committed:
Updating:
- test names for formatter tests
- dom checks for table existence
- rename columnDefinitions to columns

iros commented:
Updated @pbeshai. ping!

pbeshai commented:
+1 Thanks @iros!
